orchagent/aclorch.cpp

```diff
@@ -1787,6 +1861,47 @@ bool AclOrch::processPorts(string portsList, std::function<void (sai_object_id_t
     return true;
 }
 
+bool AclOrch::processSinglePort(AclTable &aclTable, string portAlias, std::function<void (sai_object_id_t)> inserter)
+{
+    SWSS_LOG_ENTER();
+
+    SWSS_LOG_INFO("Processing ACL table port %s", portAlias.c_str());
+
+    Port port;
+    if (!gPortsOrch->getPort(portAlias, port))
+    {
+        SWSS_LOG_WARN("Port %s not configured yet, add it to ACL table %s pending list", portAlias.c_str(), aclTable.description.c_str());
+        aclTable.pendingPortListSet.insert(portAlias);
+        return true;
+    }
+
+    switch (port.m_type)
+    {
+        case Port::PHY:
+            if (port.m_lag_member_id != SAI_NULL_OBJECT_ID)
+            {
+                SWSS_LOG_ERROR("Failed to process port. Bind table to LAG member %s is not allowed", portAlias.c_str());
+                return false;
+            }
+            inserter(port.m_port_id);
+            aclTable.bind(port.m_port_id);
+            break;
+        case Port::LAG:
+            inserter(port.m_lag_id);
+            aclTable.bind(port.m_lag_id);
+            break;
+        case Port::VLAN:
+            inserter(port.m_vlan_info.vlan_oid);
+            aclTable.bind(port.m_vlan_info.vlan_oid);
+            break;
+        default:
+            SWSS_LOG_ERROR("Failed to process port. Incorrect port %s type %d", portAlias.c_str(), port.m_type);
+            return false;
+    }
+
+    return true;
+}
+
 bool AclOrch::processAclTableType(string type, acl_table_type_t &table_type)
```

Comment:
>processAclTableType [](start = 14, length = 19)

Lots of repeated code, you can call one from the other.

Reply:
@qiluo-msft They do share some same code, I also have considered calling processPendingPort(previously processSinglePort) in processPorts. But if we look into the whole procedure, processPorts only "link" ports to tables and the action to "bind" table to ports are not included.  
For processPendingPort, it's dedicated to receiving the notification and handle the pending port, in this one function "link" and "bind" all performed. To keep the code more straightforward and readable I decide to have these two functions separated through will have some slight redundant code.

Reply:
you can move aclTable.bind out of the processPendingPort() function and make the code reuse. 

you can do aclTable.bind in doAclTablePortUpdateTask(). 



---
In reply to: [186911317](https://github.com/Azure/sonic-swss/pull/494#discussion_r186911317) [](ancestors = 186911317)